packages/sanity/src/_singletons/core/form/VersionContext.tsx

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line no-warning-comments
+/* TODO REMOVE THIS, THIS IS JUST A TEMPORARY FIX UNTIL STORES ARE SET UP */
+
+// eslint-disable-next-line import/consistent-type-specifier-style
+import {createContext, type ReactElement, useState} from 'react'
+
+import type {Version} from '../../../core/versions/types'
+import {LATEST} from '../../../core/versions/util/const'
+
+interface VersionContextValue {
+  currentVersion: Version
+  isDraft: boolean
+  setCurrentVersion: (version: Version) => void
+}
+
+export const VersionContext = createContext<VersionContextValue>({
+  currentVersion: LATEST,
+  isDraft: true,
+  // eslint-disable-next-line no-empty-function
+  setCurrentVersion: () => {},
+})
+
+interface VersionProviderProps {
+  children: ReactElement
+}
+
+export function VersionProvider({children}: VersionProviderProps): JSX.Element {
+  const [currentVersion, setCurrentVersion] = useState<Version>(LATEST)
```

Comment:
I know you are waiting here to have the store for this, but will we need a state to handle this ?
I think it could be derived from the router.

We could do something like:

```ts
export function VersionProvider({children}: VersionProviderProps): JSX.Element {
  const router = useRouter()
  const setCurrentVersion = (version: Version) => {
    const {name} = version
    if (name === 'drafts') {
      router.navigateStickyParam('perspective', '')
    } else {
      router.navigateStickyParam('perspective', `bundle.${name}`)
    }
  }
  const selectedVersion = router.stickyParams?.perspective
    ? BUNDLES.find((bundle) => bundle.name === router.stickyParams.perspective)
    : LATEST

  const currentVersion = selectedVersion || LATEST

  const isDraft = currentVersion.name === 'draft'

  const contextValue: VersionContextValue = {
    isDraft,
    setCurrentVersion,
    currentVersion,
  }

  return <VersionContext.Provider value={contextValue}>{children}</VersionContext.Provider>
}
```

To navigate and set the current version in the same action.

Reply:
We likely won't need this provider / context at all (hence why it's so rough) once we have the store :) 
It can be done to use your suggestion of course (or similar to match the rest of the context and provider in the studio) but my first thought was that we would eventually not need it.

Hence why my comment at the top of this file 👇 🙂 

``// eslint-disable-next-line no-warning-comments
/* TODO REMOVE THIS, THIS IS JUST A TEMPORARY FIX UNTIL STORES ARE SET UP */``

I can make the suggested changes as temporary things until we see if we need it since I am going to be moving towards this solution, but just a headsup given the comment I left on the file and where my mind was at :)